From 75063c00d98b64a33e9c442c336f6a804ad2512e Mon Sep 17 00:00:00 2001
From: Lucas De Marchi <lucas.demarchi@intel.com>
Date: Mon, 8 Oct 2018 16:01:57 -0700
Subject: [PATCH] Allow to run qemu >= 2.12

While trying to call "mkosi qemu" on a Clear Linux host it reported

    qemu: could not load PC BIOS '/usr/share/qemu/OVMF_CODE.fd'

Cross checking with my Fedora host the only real difference I see is the
qemu version. Changing it to pass a pflash drive directly makes it work
and also make it explicit that the it is a readonly device.
---
 mkosi | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mkosi b/mkosi
index e06f104..b22e4bf 100755
--- a/mkosi
+++ b/mkosi
@@ -3618,9 +3618,9 @@ def run_qemu(args):
     else:
         die("Couldn't find OVMF UEFI firmware blob.")
 
-    cmdline += [ "-bios", firmware,
-                 "-smp", "2",
+    cmdline += [ "-smp", "2",
                  "-m", "1024",
+                 "-drive", "if=pflash,format=raw,readonly,file=" + firmware,
                  "-drive", "format=raw,file=" + args.output,
                  *args.cmdline ]
 
-- 
2.19.1

